File: swagger/src/main/java/com/example/swagger/domain/feed/service/DeleteFeedService.java
package com.example.swagger.domain.feed.service;

import com.example.swagger.domain.feed.domain.Feed;
import com.example.swagger.domain.feed.domain.repository.FeedRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Service
public class DeleteFeedService {

    private final FeedRepository feedRepository;

    @Transactional
    public void execute(Integer feedId) {

        Feed feed = feedRepository.findById(feedId)
                .orElseThrow(() -> new RuntimeException("게시글 찾지 못함"));

        feedRepository.delete(feed);
    }

}

File: practice-kotlin/src/main/kotlin/com/example/practicekotlin/domain/feed/service/DeleteFeedService.kt
package com.example.practicekotlin.domain.feed.service

import com.example.practicekotlin.domain.feed.domain.repository.FeedRepository
import com.example.practicekotlin.domain.feed.facade.FeedFacade
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class DeleteFeedService(
    private val feedRepository: FeedRepository,
    private val feedFacade: FeedFacade
) {

    @Transactional
    fun execute(feedId: Long) {

        val feed = feedFacade.getFeedById(feedId)

        feedRepository.delete(feed)
    }

}